api/utils/clientutils/resources.go

```diff
@@ -0,0 +1,53 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package clientutils
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/defaults"
+)
+
+// IterateResources is a helper that iterates through each resource from all
+// pages and passes them one by one to the provided callback.
+func IterateResources[T any](
```

Comment:
Should we make this a proper go iterator and return an `iter.Seq` instead of a manual loop with a callback?

Reply:
also need to bump go.mod for `api/`

Reply:
@zmb3 @rosstimothy this is how imagined the iterator version. Let me know if you prefer that over this PR and i can push it here:
https://github.com/gravitational/teleport/commit/4ecd694f57fc1a3f100f1f4bc67bdee56cde1985#diff-44eb64101a2a166c4fb093edec646884a14091b1cfe42adb9cbb67a53bfc17d1

Reply:
I like it, but maybe we hold off on that for now to avoid bumping Go for /api. We can stick with what you've got in this PR.